var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -381,7 +382,8 @@ class PyTensorflow(Package, CudaPackage, ROCmPackage, PythonExtension):
 
     # https://www.tensorflow.org/install/source#tested_build_configurations
     # https://github.com/tensorflow/tensorflow/issues/70199
-    conflicts("%gcc", when="@2.17:")
+    # (-mavx512fp16 exists in gcc@12:)
+    conflicts("%gcc@:11", when="@2.17:")
```

Comment:
Our CI images only have GCC 11, so we may need to update the images if we want to be able to test future TF versions.

Reply:
I solidly think we need to be on ~12.4.0 for CI - this should also make some things easier for aarch64. I also think that it should be a spack-built gcc/clang/etc...

Reply:
I was hoping to keep things simple (no extra installs required) and representative (the most likely compiler version used by users) but I think you're probably right, especially for aarch64. Any idea how to do that? I created the current image but it was mostly by dumb luck.

Reply:
Oh, I think I can just create one for the new Ubuntu 24 release that will have newer GCC, let me do that.